src/main/java/uk/ac/ic/wlgitbridge/bridge/snapshot/SnapshotApiFacade.java

```diff
@@ -44,6 +44,15 @@ public boolean projectExists(
         }
     }
 
+    public Optional<GetDocResult> getDoc(
+            Optional<Credential> oauth2,
+            String projectName
+    ) throws FailedConnectionException, GitUserException {
+        GetDocResult doc = SnapshotApi
+                .getResult(api.getDoc(oauth2, projectName));
```

Comment:
This may be a bit tricky... It looks like `projectExists` was calling `GetDocResult.getVersionID()`, which can in some cases throw an exception:

https://github.com/winstonli/writelatex-git-bridge/blob/01d7ad8f1d4382d4f30e77d4e8e239a346d33d07/src/main/java/uk/ac/ic/wlgitbridge/snapshot/getdoc/GetDocResult.java#L110-L115

It would then catch `InvalidProjectException`, which appears to be what we get when there is a 404, and returns false.

It may be necessary to replicate that pattern here (unless we refactor GetDocResult so that it throws the appropriate exception earlier; I am not sure what the reason for not throwing the exception earlier is, but I am guessing that `fromJSON` isn't allowed to throw).

I can't currently see any path on which `doc` would actually be `null` here.

On the bright side (maybe), I can't see anywhere else that `projectExists` is used (other than in test), so we shouldn't need both methods.